circleci.go

```diff
@@ -396,6 +397,36 @@ func (c *Client) BuildOpts(account, repo, branch string, opts map[string]interfa
 	return build, nil
 }
 
+// BuildByProject triggers a build by project (this is the only way to trigger a build for project using Circle 2.1)
+func (c *Client) BuildByProject(vcsType VcsType, account string, repo string, branch string, revision string, tag string) error {
+	if tag != "" && (branch != "" || revision != "") {
+		return errors.New("cannot specify tag parameter alongside branch or revision")
+	}
+
+	resp := &BuildByProjectResponse{}
+
+	opts := map[string]interface{}{}
+	if branch != "" {
+		opts["branch"] = branch
+	}
+	if revision != "" {
+		opts["revision"] = revision
+	}
+	if tag != "" {
+		opts["tag"] = tag
+	}
+
+	err := c.request("POST", fmt.Sprintf("project/%s/%s/%s/build", vcsType, account, repo), resp, nil, opts)
```

Comment:
FYI - documentation is here: https://circleci.com/docs/api/v1-reference/#new-project-build